### DISCUSSION TITLE
Pass props through Link

### DISCUSSION THREAD

**Discussion by bleucitron** [vercel/next.js#11327] [Help]
Is it possible to pass props to a component targeted by `next/link` in a similar way to React-Router's Link ?

They have an example here: https://tylermcginnis.com/react-router-pass-props-to-link/.

**Comment by lfades** [comment]
Use [useRouter](https://nextjs.org/docs/api-reference/next/router#userouter) to get access to the context of the current route in your component.

`Link` doesn't accept custom props, it won't pass down custom props except for `href`, which is sent down when you add the `passHref` prop to `Link`.

**Reply by zontyp** [reply]
example is answer by drew reese on below link 
https://stackoverflow.com/questions/65758812/next-js-keep-the-state-or-send-props-when-router-push
